crates/ntx-builder/src/transaction.rs

```diff
@@ -145,19 +150,25 @@ impl NtxContext {
             .await
         {
             Ok(NoteAccountExecution::Success) => notes,
-            Ok(NoteAccountExecution::Failure { successful_notes, error, .. }) => {
-                let notes = successful_notes
-                    .into_iter()
-                    .map(|id| notes.iter().find(|note| note.id() == id).unwrap())
-                    .cloned()
-                    .collect::<Vec<InputNote>>();
-
-                if notes.is_empty() {
-                    let err = error.map_or_else(|| "None".to_string(), |err| err.as_report());
-                    tracing::warn!(%err, "all network notes failed");
+            Ok(NoteAccountExecution::Failure {
+                successful_notes, error, failed_note_id, ..
+            }) => {
+                if let Some(error) = error {
+                    return Err(NtxError::NoteFilter(error));
                 }
+                return Err(NtxError::ConsumptionCheck { successful_notes, failed_note_id });
```

Comment:
If the error happens, this would not produce a transaction, right? I think the behavior we want is this:

- If there is no error, we build a transaction (like we do now, I think).
- If there is an error, we mark the failed note with the error, and
  - If there are any successful notes, still build a transaction with them.
  - The remaining notes should not be affected.

Also, it may make sense to address https://github.com/0xMiden/miden-base/issues/1566 first (should be pretty simple) - let me know if you'd like me to assign it to you.

Reply:
> let me know if you'd like me to assign it to you.
Yes please.

Reply:
> If the error happens, this would not produce a transaction, right? I think the behavior we want is this:

```rust
                            self.filter_notes(&data_store, notes)
                                .and_then(|notes| self.execute(&data_store, notes))
                                .and_then(|tx| self.prove(tx))
                                .and_then(|tx| self.submit(tx))
                                .await
```

Everything before the failed note should have succeeded here ^.